gunicorn/util.py

```diff
@@ -232,33 +232,29 @@ def is_ipv6(addr):
     return True
 
 
-def parse_address(netloc, default_port=8000):
+def parse_address(netloc, default_port='8000'):
+    host, port = [netloc, default_port]
+
     if re.match(r'unix:(//)?', netloc):
         return re.split(r'unix:(//)?', netloc)[-1]
 
     if netloc.startswith("tcp://"):
         netloc = netloc.split("tcp://")[1]
 
-    # get host
     if '[' in netloc and ']' in netloc:
-        host = netloc.split(']')[0][1:].lower()
+        host = netloc.split(']')[0][1:]
+        port = (netloc.split(']:') + [default_port])[-1]
```

Comment:
I feel like this should be `[1]` instead of `[-1]`? We want _either_ the port that was specified in the netloc or the default port, right?

If we're rewriting this code I think we can make it less tricky by being a bit clearer about this stuff.